scilpy/reconst/multi_processes.py

```diff
@@ -0,0 +1,420 @@
+                           itertools.repeat(B),
+                           itertools.repeat(sphere),
+                           itertools.repeat(relative_peak_threshold),
+                           itertools.repeat(absolute_threshold),
+                           itertools.repeat(min_separation_angle),
+                           itertools.repeat(npeaks),
+                           itertools.repeat(normalize_peaks),
+                           np.arange(len(chunks))))
+    pool.close()
+    pool.join()
+
+    peak_dirs_array = np.zeros((np.prod(data_shape[0:3]), npeaks, 3))
+    peak_values_array = np.zeros((np.prod(data_shape[0:3]), npeaks))
+    peak_indices_array = np.zeros((np.prod(data_shape[0:3]), npeaks))
+    for i, peak_dirs, peak_values, peak_indices in results:
+        peak_dirs_array[chunk_len[i]:chunk_len[i+1], :] = peak_dirs
+        peak_values_array[chunk_len[i]:chunk_len[i+1], :] = peak_values
+        peak_indices_array[chunk_len[i]:chunk_len[i+1], :] = peak_indices
+
+    peak_dirs_array = peak_dirs_array.reshape(data_shape[0:3]+(npeaks, 3))
+    peak_values_array = peak_values_array.reshape(data_shape[0:3]+(npeaks,))
+    peak_indices_array = peak_indices_array.reshape(data_shape[0:3]+(npeaks,))
+
+    return peak_dirs_array, peak_values_array, peak_indices_array
+
+
+def maps_from_sh_parallel(args):
+    shm_coeff = args[0]
+    peak_dirs = args[1]
+    peak_values = args[2]
+    peak_indices = args[3]
+    B = args[4]
+    sphere = args[5]
+    gfa_thr = args[6]
+    chunk_id = args[7]
+
+    data_shape = shm_coeff.shape[0]
+    nufo_map = np.zeros(data_shape)
+    afd_map = np.zeros(data_shape)
+    afd_sum = np.zeros(data_shape)
+    rgb_map = np.zeros((data_shape, 3))
+    gfa_map = np.zeros(data_shape)
+    qa_map = np.zeros((data_shape, peak_values.shape[1]))
+
+    max_odf = 0
+    global_max = -np.inf
+    for idx in range(len(shm_coeff)):
+        if not np.isnan(shm_coeff[idx]).any():
+            odf = np.dot(shm_coeff[idx], B)
+            odf = odf.clip(min=0)
+            sum_odf = np.sum(odf)
+            max_odf = np.maximum(max_odf, sum_odf)
+            if sum_odf > 0:
+                rgb_map[idx] = np.dot(np.abs(sphere.vertices).T, odf)
+                rgb_map[idx] /= np.linalg.norm(rgb_map[idx])
+                rgb_map[idx] *= sum_odf
+            gfa_map[idx] = gfa(odf)
+            if gfa_map[idx] < gfa_thr:
+                global_max = max(global_max, odf.max())
+            elif np.sum(peak_indices[idx] > -1):
+                nufo_map[idx] = np.sum(peak_indices[idx] > -1)
+                afd_map[idx] = peak_values[idx].max()
+                afd_sum[idx] = np.sqrt(np.dot(shm_coeff[idx], shm_coeff[idx]))
+                qa_map = peak_values[idx] - odf.min()
+                global_max = max(global_max, peak_values[idx][0])
+
+    rgb_map /= max_odf
+    rgb_map *= 255
+    qa_map /= global_max
+
+    return chunk_id, nufo_map, afd_map, afd_sum, rgb_map, gfa_map, qa_map
```

Comment:
Wasn't there 3 afd maps ? Max, Total, Sum ?

Reply:
If you look at the old scil_compute_fodf_metrics.py, I think there is only 2 afd maps.

Reply:
My bad, there is in fact 3 output files. I will add it!

Reply:
But should it come from this function or be calculated in scil_compute_fodf_metrics.py, since it is just data[:, :, :, 0].

Reply:
I already had it in scil_compute_fodf_metrics.py

Reply:
I would need a second opinion on that.
@GuillaumeTh do you consider doing data[:, :, :, 0] a 'metrics' and should it be inside the function metrics_from_sh?

I would say yes, but it does seems overly simple

Reply:
So what about that? @frheault  @GuillaumeTh